pkg/provider/cloud/openstack/helper.go

```diff
@@ -468,7 +468,7 @@ func getSubnetForNetwork(netClient *gophercloud.ServiceClient, networkIDOrName s
 }
 
 func isNotFoundErr(err error) bool {
-	if _, ok := err.(gophercloud.ErrDefault404); ok || strings.Contains(err.Error(), "not found") {
+	if _, ok := err.(gophercloud.ErrDefault404); ok || strings.Contains(errors.Unwrap(err).Error(), "not found") {
```

Comment:
I'm not fond of the `string.Contains`, but this is not something you introduced. On the other hand, the type check should do the unwrapping and not the string match. You can use 

```suggestion
         var errNotFound gophercloud.ErrDefault404
	if errors.Is(err, &errNotFound); ok || strings.Contains(errors.Error(), "not found") {
```

Reply:
sure